src/lib/core/utils/show-hide-variable-context.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { createContext, useState } from 'react';
+
+export const ShowHideVariableContext = createContext({
+  toggleShowHideVariable: false,
+  setToggleShowHideVariableHandler: (toggleShowHideVariable: boolean) => {},
```

Comment:
I think these could be named a little better. The current names are a bit ambiguous. How about `showOnlyCompatibleVariables`/`setShowOnlyCompatibleVariables`?